python/semantic_kernel/kernel.py

```diff
@@ -286,38 +291,91 @@ async def run_async(
 
         pipeline_step = 0
         for func in functions:
-            assert isinstance(func, SKFunctionBase), (
-                "All func arguments to Kernel.run*(inputs, func1, func2, ...) "
-                "must be SKFunctionBase instances"
-            )
-
-            if context.error_occurred:
-                self._log.error(
-                    f"Something went wrong in pipeline step {pipeline_step}. "
-                    f"Error description: '{context.last_error_description}'"
+            while True:
+                assert isinstance(func, SKFunctionBase), (
+                    "All func arguments to Kernel.run*(inputs, func1, func2, ...) "
+                    "must be SKFunctionBase instances"
                 )
-                return context
 
-            pipeline_step += 1
+                if context.error_occurred:
+                    self._log.error(
+                        f"Something went wrong in pipeline step {pipeline_step}. "
+                        f"Error description: '{context.last_error_description}'"
+                    )
+                    return context
 
-            try:
-                context = await func.invoke_async(input=None, context=context, **kwargs)
+                pipeline_step += 1
```

Comment:
As we added the functionality to request repeat, I believe this counter should go at the end of function execution. Because currently, if you request repeat for the function, the value of counter will be incremented, but it should be the same, since you will re-run the same function/pipeline step.